trulens_eval/trulens_eval/utils/generated.py

```diff
@@ -65,14 +65,16 @@ def re_0_10_rating(s: str) -> int:
         ParseError: If no integers between 0 and 10 are found in the string.
     """
 
-    matches = PATTERN_INTEGER.findall(s)
+    matches = PATTERN_NUMBER.findall(s)
     if not matches:
-        raise ParseError("int or float number", s, pattern=PATTERN_INTEGER)
+        raise ParseError("int or float number", s, pattern=PATTERN_NUMBER)
 
     vals = set()
     for match in matches:
         try:
-            vals.add(validate_rating(int(match)))
+            vals.add(
+                validate_rating(int(float(match)))
```

Comment:
kinda late to the game, but:
1. Why are we constrained to ints?
2. If we are constrained to ints, shouldn't we round it instead of flooring it?
3. The doc on L54 says "If the string does not match an integer ... raises an error ...", but this won't do that.

Reply:
1. No particular reason beyond easier intrepretability for the end users AFAIK.
2. Make sense - I can make a change to this
3. Same as 1. - I do recall we went back and forth a bit on this and the doc just became outdated b/c of my change. Will update

Reply:
https://github.com/truera/trulens/pull/1244 fix PR